File: app/src/main/java/com/anokmik/databinding/communicator/NotifyGreetingFragmentCommunicator.java
package com.anokmik.databinding.communicator;

import android.databinding.adapters.TextViewBindingAdapter;

import com.anokmik.databinding.model.NotifyGreeting;

public class NotifyGreetingFragmentCommunicator {

    public NotifyGreeting notifyGreeting = new NotifyGreeting();

    public TextViewBindingAdapter.OnTextChanged onGreetingTextChanged = new TextViewBindingAdapter.OnTextChanged() {
        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
            notifyGreeting.setName(s.toString());
        }
    };

}

File: app/src/main/kotlin/com/anokmik/databinding/communicator/NotifyGreetingFragmentCommunicator.kt
package com.anokmik.databinding.communicator;

import android.databinding.adapters.TextViewBindingAdapter
import com.anokmik.databinding.model.NotifyGreeting

class NotifyGreetingFragmentCommunicator {

    @JvmField
    val notifyGreeting: NotifyGreeting = NotifyGreeting()

    @JvmField
    val onGreetingTextChanged: TextViewBindingAdapter.OnTextChanged = TextViewBindingAdapter.OnTextChanged { s, start, before, count -> notifyGreeting.name = s.toString() }

}